util/src/main/java/tech/pegasys/pantheon/util/enode/EnodeURL.java

```diff
@@ -127,18 +125,43 @@ private static void checkStringArgumentNotEmpty(final String argument, final Str
     checkArgument(argument != null && !argument.trim().isEmpty(), message);
   }
 
+  public static boolean sameListeningEndpoint(final EnodeURL enodeA, final EnodeURL enodeB) {
+    if (enodeA == null || enodeB == null) {
+      return false;
+    }
+
+    return Objects.equal(enodeA.nodeId, enodeB.nodeId)
```

Comment:
Google's Objects.equal says not to use it after Java 7.  Can these change to java.util.Object.equals?  (the replacement Guava recommends)

Reply:
yikes - we have a lot of these (Objects.hashCode too), wonder if we can we add an error-prone check for this.  I'll open a ticket to clean this up globally.

Reply:
I've got a PR ready to roll that conflicts with some of this code.  If you want to submit it as is that's fine, I can get the PR out today.

Reply:
https://pegasys1.atlassian.net/browse/PAN-2629